raco/backends/myria/myria.py

```diff
@@ -1937,51 +1952,50 @@ def fire(self, op):
             op.args[idx] = producer
         return op
 
+    def __str__(self):
+        return ("ShuffleConsumer(ShuffleProducer(SplitConsumer(SplitProducer"
+                "(input)))) -> ShuffleConsumer(ShuffleProducer(input))")
+
 
 class DoUntilConvergence(rules.Rule):
 
     @staticmethod
-    def replace_scan_with_idb_consumer(expr, idb_controllers, idb_producers):
-        if isinstance(expr, algebra.ZeroaryOperator):
+    def replace_scan_with_idb_consumer(op, idb_controllers, idb_producers):
+        if isinstance(op, algebra.ZeroaryOperator):
             return
-        for idx, op in enumerate(expr.children()):
-            if isinstance(op, algebra.ScanIDB):
-                if op.name not in idb_producers:
-                    idb_producers[op.name] = \
-                        MyriaSplitProducer(idb_controllers[op.name])
-                consumer = MyriaConsumer(None, idb_producers[op.name])
-                if isinstance(expr, algebra.UnaryOperator):
-                    expr.input = consumer
-                elif isinstance(expr, algebra.BinaryOperator):
-                    if idx == 0:
-                        expr.left = consumer
-                    else:
-                        expr.right = consumer
-                elif isinstance(expr, algebra.NaryOperator):
-                    expr.args[idx] = consumer
-                if isinstance(expr, algebra.Select) and \
-                        isinstance(expr.condition, expression.boolean.GTEQ):
-                    (group_list, agg) = \
-                        idb_controllers[op.name].get_group_agg()
+        for ch in op.children():
+            if isinstance(ch, algebra.ScanIDB):
+                if ch.name not in idb_producers:
+                    idb_producers[ch.name] = MyriaSplitProducer(
+                        idb_controllers[ch.name])
+                consumer = MyriaConsumer(None, idb_producers[ch.name])
+                replace_child_with(op, ch, consumer)
+                if (isinstance(op, algebra.Select) and
+                        (isinstance(op.condition, expression.boolean.GTEQ) or
+                         isinstance(op.condition, expression.boolean.GT))):
+                    group_list, agg = idb_controllers[ch.name].get_group_agg()
                     if isinstance(agg, expression.aggregate.COUNTALL):
-                        agg.threshold = expr.condition.right.value
-                        expr.condition = None
+                        if isinstance(op.condition, expression.boolean.GTEQ):
+                            agg.threshold = op.condition.right.value
+                        else:
+                            # count generates integers, translate it to GTEQ
+                            agg.threshold = op.condition.right.value + 1
```

Comment:
The if condition above ensures that it's either a `GTEQ` or `GT` so I put `else` here.